Proposal for a Usage section in Readme

Given most examples I saw in https://github.com/bjartek/go-with-the-flow/network/dependents?package_id=UGFja2FnZS0yMjc1NjE0OTAz use v1, I thought maybe it could be useful to add a `## Usage` section to the read me that shows how to get setup with v2.

I leave my proposed Usage section below.

## Usage

First create a project directory, initialise the go module and install go-with-the-flow :
  ```sh
  mkdir test-gwtf && cd test-gtwf
  flow init
  go mod init github.com/louisguitton/test-gwtf
  go get github.com/bjartek/go-with-the-flow/v2/gwtf
  ```
Then create a task file:
  ```sh
  touch tasks/main.go
  ```
In that task file, you can then import gwtf and use it to your convenience, for example:
  ```go
  package main

  import (
	  "github.com/bjartek/go-with-the-flow/v2/gwtf"
  )
  
  func main() {
	  flow := gwtf.NewGoWithTheFlowInMemoryEmulator()
  
	  flow.CreateAccounts("0x02")
  }
  ```

In this occasion, we want to create an account called `0x02` so we need to edit `flow.json` to add that account to the accounts array:
```json
		"0x02": {
			"address": "01cf0e2f2f715450",
			"key": "62fc6b3839ebc9265ee5eae45191d6efcd60fcf143a936fc3db49312eb0f2ab6"
		}
```

Then you can run in one tab
```sh
flow emulator -v
```
and in another tab
```sh
go run ./tasks/main.go
```

the main issue that I faced while getting setup was that I had `github.com/onflow/flow-go-sdk v0.24.0` previously installed and listed in my `go.mod`, and then I was getting the following error when running the minimal example:

```sh
→ go get github.com/bjartek/go-with-the-flow/v2/gwtf
# github.com/onflow/flow-cli/pkg/flowkit
../../../go/pkg/mod/github.com/onflow/flow-cli@v0.28.3/pkg/flowkit/keys.go:119:49: too many arguments in call to kmsClient.SignerForKey
	have (context.Context, flow.Address, cloudkms.Key)
	want (context.Context, cloudkms.Key)
../../../go/pkg/mod/github.com/onflow/flow-cli@v0.28.3/pkg/flowkit/transaction.go:108:14: assignment mismatch: 1 variable but cadence.NewString returns 2 values
../../../go/pkg/mod/github.com/onflow/flow-cli@v0.28.3/pkg/flowkit/transaction.go:109:14: assignment mismatch: 1 variable but cadence.NewString returns 2 values
# github.com/onflow/flow-go/fvm
../../../go/pkg/mod/github.com/onflow/flow-go@v0.18.4/fvm/env.go:32:5: cannot use &hostEnv{} (type *hostEnv) as type runtime.Interface in assignment:
	*hostEnv does not implement runtime.Interface (missing AllocateStorageIndex method)
../../../go/pkg/mod/github.com/onflow/flow-go@v0.18.4/fvm/env.go:138:19: cannot use e (type *hostEnv) as type runtime.Interface in field value:
	*hostEnv does not implement runtime.Interface (missing AllocateStorageIndex method)
../../../go/pkg/mod/github.com/onflow/flow-go@v0.18.4/fvm/script.go:84:4: cannot use env (type *hostEnv) as type runtime.Interface in field value:
	*hostEnv does not implement runtime.Interface (missing AllocateStorageIndex method)
../../../go/pkg/mod/github.com/onflow/flow-go@v0.18.4/fvm/transactionContractFunctionInvocator.go:62:4: cannot use env (type *hostEnv) as type runtime.Interface in field value:
	*hostEnv does not implement runtime.Interface (missing AllocateStorageIndex method)
../../../go/pkg/mod/github.com/onflow/flow-go@v0.18.4/fvm/transactionInvocator.go:135:5: cannot use env (type *hostEnv) as type runtime.Interface in field value:
	*hostEnv does not implement runtime.Interface (missing AllocateStorageIndex method)
../../../go/pkg/mod/github.com/onflow/flow-go@v0.18.4/fvm/transactionInvocator.go:277:43: not enough arguments in call to interpreter.NewHostFunctionValue
	have (func(interpreter.Invocation) interpreter.Value)
	want (interpreter.HostFunction, *sema.FunctionType)
```

Tracing that down to the fact I should have `github.com/onflow/flow-go-sdk v0.20.1-0.20210623043139-533a95abf071` took me some time, and a Usage section would have helped me I think.

--- a/Makefile
+++ b/Makefile
@@ -4,10 +4,10 @@ coveralls:
 	go test -coverprofile=profile.cov -covermode=atomic -coverpkg=github.com/bjartek/go-with-the-flow/v2/gwtf -v ./...
 
 cover: test
-	 go tool cover -html=profile.cov
+	go tool cover -html=profile.cov
 
 install-gotestsum:
-	 go get gotest.tools/gotestsum
+	go get gotest.tools/gotestsum
 
 test-report: install-gotestsum
 	gotestsum -f testname --no-color --hide-summary failed --junitfile test-result.xml

--- a/README.md
+++ b/README.md
@@ -1,42 +1,49 @@
-[![Coverage Status](https://coveralls.io/repos/github/bjartek/go-with-the-flow/badge.svg?branch=main)](https://coveralls.io/github/bjartek/go-with-the-flow?branch=main) [![ci](https://github.com/bjartek/go-with-the-flow/actions/workflows/test.yml/badge.svg)](https://github.com/bjartek/go-with-the-flow/actions/workflows/test.yml)
+[![Coverage Status](https://coveralls.io/repos/github/bjartek/go-with-the-flow/badge.svg?branch=main)](https://coveralls.io/github/bjartek/go-with-the-flow?branch=main) [![ci](https://github.com/bjartek/go-with-the-flow/actions/workflows/ci.yml/badge.svg)](https://github.com/bjartek/go-with-the-flow/actions/workflows/ci.yml)
 
 # Go with the Flow
 
-Set of go scripts to make it easer to run a Story consisting of creating accounts, deploying contracts, executing transactions and running scripts on the Flow Blockchain.
+> Tooling to help develop application on the the Flow Blockchain
 
-Feel free to ask questions to @bjartek in the Flow Discord.
+Set of go scripts to make it easer to run a story consisting of creating accounts,
+deploying contracts, executing transactions and running scripts on the Flow Blockchain.
+These go scripts also make writing integration tests of your smart contracts much easier.
+
+## Information
+
+### Main features
+
+- Create a single go file that will start emulator, deploy contracts, create accounts and run scripts and transactions. see `examples/demo/main.go`
+- Fetch events, store progress in a file and send results to Discord. see `examples/event/main.go`
+- Support inline scripts if you do not want to store everything in a file when testing
+- Supports writing tests against transactions and scripts with some limitations on how to implement them.
+- Asserts to make it easier to use the library in writing tests see `examples/transaction_test.go` for examples
+
+### Gotchas
+
+- When specifying extra accounts that are created on emulator they are created in alphabetical order, the addresses the emulator assign is always fixed.
+- tldr; Name your stakeholder accounts in alphabetical order
+
+### Note on v2
 
 v2 of GoWithTheFlow removed a lot of the code in favor of `flowkit` in the flow-cli. Some of the code from here was
 contributed by me into flow-cli like the goroutine based event fetcher.
 
 Breaking changes between v1 and v2:
- - v1 had a config section for discord webhooks. That has been removed since the flow-cli will remove extra config things in flow.json. Store the webhook url in an env variable and use it as argument when creating the DiscordWebhook struct.
+
+- v1 had a config section for discord webhooks. That has been removed since the flow-cli will remove extra config things in flow.json. Store the webhook url in an env variable and use it as argument when creating the DiscordWebhook struct.
 
 Special thanks to @sideninja for helping me get my changes into flow-cli. and for jayShen that helped with fixing some issues!
 
-## Main features
- - Create a single go file that will start emulator, deploy contracts, create accounts and run scripts and transactions. see `examples/demo/main.go` 
- - Fetch events, store progress in a file and send results to Discord. see `examples/event/main.go`
- - Support inline scripts if you do not want to sture everything in a file when testing 
- - Supports writing tests against transactions and scripts with some limitations on how to implement them. 
- - Asserts to make it easier to use the library in writing tests see `examples/transaction_test.go` for examples
+## Resources
 
-## Gotchas
- - When specifying extra accounts that are created on emulator they are created in alphabetical order, the addresses the emulator assign is always fixed. 
- - tldr; Name your stakeholder acounts in alphabetical order
-
-## Examples
-
-In order to run the demo example you only have to run `make` in the example folder of this project. 
-The emulator will be run in memory. 
+- Run the demo example in this project with `cd example && make`. The emulator will be run in memory.
+- Check [other codebases](https://github.com/bjartek/go-with-the-flow/network/dependents?package_id=UGFja2FnZS0yMjc1NjE0OTAz) that use this project
+- Feel free to ask questions to @bjartek in the Flow Discord.
 
 ## Credits
 
-This project is a rewrite of https://github.com/versus-flow/go-flow-tooling 
+This project is a rewrite of https://github.com/versus-flow/go-flow-tooling
 
+## Todo
 
-## Todo 
- - fix golangci-rules, disabled a bunch to get the build green
-
-
-
+- fix golangci-rules, disabled a bunch to get the build green

--- a/README.md
+++ b/README.md
@@ -22,6 +22,9 @@ These go scripts also make writing integration tests of your smart contracts muc
 
 - When specifying extra accounts that are created on emulator they are created in alphabetical order, the addresses the emulator assign is always fixed.
 - tldr; Name your stakeholder accounts in alphabetical order
+- When writing integration tests, tests must be in the same folder as flow.json
+with contracts and transactions/scripts in subdirectories in order for the path resolver
+to work correctly
 
 ### Note on v2
 
@@ -39,6 +42,48 @@ Special thanks to @sideninja for helping me get my changes into flow-cli. and fo
 - Run the demo example in this project with `cd example && make`. The emulator will be run in memory.
 - Check [other codebases](https://github.com/bjartek/go-with-the-flow/network/dependents?package_id=UGFja2FnZS0yMjc1NjE0OTAz) that use this project
 - Feel free to ask questions to @bjartek in the Flow Discord.
+
+## Usage
+
+First create a project directory, initialize the go module and install `go-with-the-flow`:
+
+```sh
+mkdir test-gwtf && cd test-gtwf
+flow init
+go mod init example.com/test-gwtf
+go get github.com/bjartek/go-with-the-flow/v2/gwtf
+```
+
+Then create a task file:
+
+```sh
+touch tasks/main.go
+```
+
+In that task file, you can then import `go-with-the-flow` and use it to your convenience, for example:
+
+```go
+package main
+
+import (
+    "fmt"
+
+    "github.com/bjartek/go-with-the-flow/v2/gwtf"
+)
+
+func main() {
+    g := gwtf.NewGoWithTheFlowInMemoryEmulator()
+    fmt.Printf("%v", g.State.Accounts())
+}
+```
+
+Then you can run
+
+```sh
+go run ./tasks/main.go
+```
+
+This is a minimal example that only prints accounts, but from there you can branch out.
 
 ## Credits
 

--- a/examples/event_integration_test.go
+++ b/examples/event_integration_test.go
@@ -22,7 +22,7 @@ func TestEvents(t *testing.T) {
 
 		_, err := g.EventFetcher().End(2).From(-10).Event("A.0ae53cb6e3f42a79.FlowToken.TokensMinted").Run()
 		assert.Error(t, err)
-		assert.Contains(t, "FromIndex is negative", err.Error())
+		assert.Contains(t, err.Error(), "FromIndex is negative")
 	})
 
 	t.Run("Fetch last events", func(t *testing.T) {

--- a/examples/script_integration_test.go
+++ b/examples/script_integration_test.go
@@ -12,12 +12,12 @@ func TestScript(t *testing.T) {
 
 	t.Run("Raw account argument", func(t *testing.T) {
 		value := g.ScriptFromFile("test").RawAccountArgument("0x1cf0e2f2f715450").RunReturnsInterface()
-		assert.Equal(t, value, "0x1cf0e2f2f715450")
+		assert.Equal(t, "0x1cf0e2f2f715450", value)
 	})
 
 	t.Run("Raw account argument", func(t *testing.T) {
 		value := g.ScriptFromFile("test").AccountArgument("first").RunReturnsInterface()
-		assert.Equal(t, value, "0x1cf0e2f2f715450")
+		assert.Equal(t, "0x1cf0e2f2f715450", value)
 	})
 
 	t.Run("Script should report failure", func(t *testing.T) {

--- a/examples/setup_integration_test.go
+++ b/examples/setup_integration_test.go
@@ -10,27 +10,27 @@ func TestSetupIntegration(t *testing.T) {
 
 	t.Run("Should create inmemory emulator client", func(t *testing.T) {
 		g := gwtf.NewGoWithTheFlowInMemoryEmulator()
-		assert.Equal(t, g.Network, "emulator")
+		assert.Equal(t, "emulator", g.Network)
 	})
 
 	t.Run("Should create local emulator client", func(t *testing.T) {
 		g := gwtf.NewGoWithTheFlowEmulator()
-		assert.Equal(t, g.Network, "emulator")
+		assert.Equal(t, "emulator", g.Network)
 	})
 
 	t.Run("Should create testnet client", func(t *testing.T) {
 		g := gwtf.NewGoWithTheFlowDevNet()
-		assert.Equal(t, g.Network, "testnet")
+		assert.Equal(t, "testnet", g.Network)
 	})
 
-	t.Run("Should create testnet client with for network metdho", func(t *testing.T) {
+	t.Run("Should create testnet client with for network method", func(t *testing.T) {
 		g := gwtf.NewGoWithTheFlowForNetwork("testnet")
-		assert.Equal(t, g.Network, "testnet")
+		assert.Equal(t, "testnet", g.Network)
 	})
 
 	t.Run("Should create mainnet client", func(t *testing.T) {
 		g := gwtf.NewGoWithTheFlowMainNet()
-		assert.Equal(t, g.Network, "mainnet")
+		assert.Equal(t, "mainnet", g.Network)
 		assert.True(t, g.PrependNetworkToAccountNames)
 		g = g.DoNotPrependNetworkToAccountNames()
 		assert.False(t, g.PrependNetworkToAccountNames)

